src/map/map.cpp

```diff
@@ -676,15 +675,25 @@ int32 send_parse(int8 *buff, size_t* buffsize, sockaddr_in* from, map_session_da
             packetList.pop_front();
             packets++;
         }
+
+        PacketCount /= 2;
+
         //Сжимаем данные без учета заголовка
         //Возвращаемый размер в 8 раз больше реальных данных
-        PacketSize = zlib_compress(buff + FFXI_HEADER_SIZE, *buffsize - FFXI_HEADER_SIZE, PTempBuff, *buffsize);
+        PacketSize = zlib_compress(buff + FFXI_HEADER_SIZE, *buffsize - FFXI_HEADER_SIZE, PTempBuff, map_config.buffer_size);
+
+        // handle compression error
+        if (PacketSize == static_cast<int32>(-1))
+        {
+            packets = 0;
```

Comment:
There will be more problems if that happens though. The code doesn't do any checking whatsoever after that so things are in pretty undefined state. There is that PTempBuf overflowed message, but it doesn't return or anything, it has already corrupted the memory and continues to corrupt memory.